tests/testthat/test-rstudio.R

```diff
@@ -0,0 +1,79 @@
+context("rstudio")
+
+test_that("use_rstudio() creates .Rproj file, named after directory", {
+  dir <- scoped_temporary_package(rstudio = FALSE)
+  capture_output(use_rstudio())
+  rproj <- list.files(proj_get(), pattern = "\\.Rproj$")
+  expect_identical(rproj, paste0(basename(dir), ".Rproj"))
+})
+
+test_that("a non-RStudio project is recognized", {
+  scoped_temporary_package(rstudio = FALSE)
+  expect_false(is_rstudio_project())
+  expect_identical(rproj_path(), NA_character_)
+})
+
+test_that("an RStudio project is recognized", {
+  scoped_temporary_package(rstudio = TRUE)
+  expect_true(is_rstudio_project())
+  expect_match(rproj_path(), "\\.Rproj$")
+})
+
+test_that("we error for multiple Rproj files", {
+  scoped_temporary_package(rstudio = TRUE)
+  file.copy(
+    file.path(proj_get(), rproj_path()),
+    file.path(proj_get(), "copy.Rproj")
+  )
+  expect_error(rproj_path(), "Multiple .Rproj files found", fixed = TRUE)
+})
+
+test_that("Rproj is parsed (actually, only colon-containing lines)", {
+  tmp <- tempfile()
+  writeLines(c("a: a", "", "b: b", "I have no colon"), tmp)
+  expect_identical(
+    parse_rproj(tmp),
+    list(a = "a", "", b = "b", "I have no colon")
+  )
+})
+
+test_that("Existing field(s) in Rproj can be modified", {
+  tmp <- tempfile()
+  writeLines(
+    c(
+      "Version: 1.0",
+      "",
+      "RestoreWorkspace: Default",
+      "SaveWorkspace: Yes",
+      "AlwaysSaveHistory: Default"
+    ),
+    tmp
+  )
+  before <- parse_rproj(tmp)
+  delta <- list(RestoreWorkspace = "No", SaveWorkspace = "No")
+  after <- modify_rproj(tmp, delta)
+  expect_identical(before[c(1, 2, 5)], after[c(1, 2, 5)])
+  expect_identical(after[3:4], delta)
+})
+
+test_that("we can roundtrip an Rproj file", {
+  scoped_temporary_package(rstudio = TRUE)
+  tmp <- tempfile()
+  rproj <- modify_rproj(
+    file.path(proj_get(), rproj_path()),
+    list()
+  )
+  writeLines(serialize_rproj(rproj), tmp)
+  expect_equivalent(
```

Comment:
Why not `expect_equal()` on the lines themselves?

Reply:
ok changed